### PR TITLE
CON-4042 Price type reset

### PR DESCRIPTION
**Pull request by mkafadarov**

### CODE REVIEW

File: Controllers/Backend/ConnectConfig.php [hunk 1]
```diff
@@ -871,6 +928,21 @@ public function getFactory()
     }
 
     /**
+     * @return string
+     */
+    protected function getHost()
```
Comment (sbofirov) [inline Controllers/Backend/ConnectConfig.php]: Please move this function in ConfigComponent. After that you can use it here and in ConnectBaseController as well.
It should be also removed from ConnectBaseController.

Modify in all places where it's called.

File: Snippets/backend/connect/view/main.ini [hunk 2]
```diff
@@ -182,6 +182,10 @@ mapping/columns/connect-category = "[0] Category"
 mapping/columns/shopware-category = "Shopware Category"
 navigation/products = "Products"
 import/toolbar/show_in_groups = "Show in groups"
+config/price_reset_btn = "reset"
+config/price_reset_label = "Reset exported prices"
+config/price_reset_message = "Your exported products will be deleted in Connect and your sent offers will be invalid.<br>Do you want to continue?"
+config/price_reset_success = "The exported prices were successfully reset. It will take up to 10min for the changes to take effect. When this operation is done, you will get the option to set the price type again when you reopen "Export" in the Connect menu."
```
Comment (sbofirov) [inline Snippets/backend/connect/view/main.ini]: Maybe double quotes ```"Export"``` will break the snippets

File: Snippets/backend/connect/view/main.ini [hunk 3]
```diff
@@ -514,6 +518,10 @@ navigation/mapping/export = "Export"
 mapping/columns/connect-category = "[0] Kategorie"
 mapping/columns/shopware-category = "Shopware Kategorie"
 navigation/products = "Produkte"
+config/price_reset_btn = "zurücksetzen"
+config/price_reset_label = "Exportierte Preise zurücksetzen"
+config/price_reset_message = "Ihre exportierten Produkte auf Connect werden gelöscht, ihre Angebote werden dadurch ungültig.<br>Möchten Sie fortfahren?"
+config/price_reset_success = "Die exportierten Preise wurden erfolgreich zurückgesetzt. Es kann bis zu 10 Minuten dauern, bis diese Änderung Wirkung zeigt. Wenn die Änderung durchgeführt wurde, haben Sie beim Öffnen des Punktes "Produkte zu Connect" die Möglichkeit die zu exportierenden Preise neu zu setzen."
```
Comment (sbofirov) [inline Snippets/backend/connect/view/main.ini]: same here

File: Views/backend/connect/controller/main.js [hunk 4]
```diff
@@ -1427,6 +1430,29 @@ Ext.define('Shopware.apps.Connect.controller.Main', {
...
+                var response = Ext.JSON.decode(result.responseText);
+                if (response.success) {
+                    me.createGrowlMessage(
+                        me.messages.priceResetLabel,
+                        me.messages.priceResetSuccess
+                    );
+                } else {
+                    me.createGrowlMessage(
+                        me.messages.priceResetLabel,
+                        response.message
```
Comment (sbofirov) [inline Views/backend/connect/controller/main.js]: This error is not translated. It comes from SN. Do we need to show some generic message and real reason to be in the logs?

File: Controllers/Backend/ConnectConfig.php (hunk 1 above)
Comment (mkafadarov) [inline Controllers/Backend/ConnectConfig.php]: Sorry about that I forgot to remove this method. I am not using it anywhere

**Pull request merged**